command_line/reindex.py

```diff
@@ -311,19 +312,17 @@ def run(args):
     else:
         change_of_basis_op = sgtbx.change_of_basis_op(params.change_of_basis_op)
 
-    # validate cb_op
-    try:
-        _ = sgtbx.rt_mx(change_of_basis_op.as_xyz())
-    except RuntimeError:
-        sys.exit("Unsuitable change of basis: %s" % params.change_of_basis_op)
-
     if len(experiments):
         space_group = params.space_group
         if space_group is not None:
             space_group = space_group.group()
-        experiments = reindex_experiments(
-            experiments, change_of_basis_op, space_group=space_group
-        )
+        try:
+            experiments = reindex_experiments(
+                experiments, change_of_basis_op, space_group=space_group
+            )
+        except RuntimeError as e:
```

Comment:
Do we want to catch _any_ `RuntimeError`, or just `"Unsuitable value for rational rotation matrix"` errors? If the latter, we could explicitly check the error message and re-raise the error if it is something different.

Reply:
Hmm, this is a good point, this would cause potentially unhelpful "Error" messages in other exception cases. Maybe we should throw/catch a special exception at the source rather than a RuntimeError? Alternatively, raise the `sys.exit()` where it happens. That case, however, makes it harder to use from other processes that might _catch_ the error.

Some combination of both?

Reply:
It turns out we do have form doing this https://github.com/dials/dials/blob/c5e652f9b01cfd666db9a51171d24dae72d76534/algorithms/indexing/symmetry.py#L106-L108
so given we can't change the exception origins (in rstbx C++) I'm reasonably happy catching the error explicitly.